an-rope/src/internals.rs

```diff
@@ -197,8 +166,7 @@ impl Node {
     fn subtree_weight (&self) -> usize {
         match self { &Leaf(ref s) => s.len()
                    , &Branch(BranchNode { ref left, .. }) =>
-                        left.as_ref().map(Box::as_ref)
-                            .map_or(0, Node::subtree_weight)
+                        left.subtree_weight()
```

Comment:
Shouldn't this be `left.len()`? Otherwise this just dives down to the leftmost leaf and returns the length of that.